app/scripts/components/analysis/define/aoi-upload-modal.tsx

```diff
@@ -0,0 +1,158 @@
+import React, { useCallback, useEffect, useRef } from 'react';
+import styled from 'styled-components';
+import { Feature } from 'geojson';
+import { Modal, ModalFooter, ModalHeadline } from '@devseed-ui/modal';
+
+import { Button } from '@devseed-ui/button';
+import { themeVal, visuallyHidden } from '@devseed-ui/theme-provider';
+import {
+  CollecticonArrowUp,
+  CollecticonTickSmall,
+  CollecticonXmarkSmall,
+  CollecticonCircleExclamation,
+  CollecticonCircleTick,
+  CollecticonCircleInformation
+} from '@devseed-ui/collecticons';
+import useCustomAoI, { acceptExtensions } from './use-custom-aoi';
+import { variableGlsp } from '$styles/variable-utils';
+
+const UploadFileModalFooter = styled(ModalFooter)`
+  display: flex;
+  justify-content: right;
+  flex-flow: row nowrap;
+  gap: ${variableGlsp(0.25)};
+`;
+
+const UploadFileModalBody = styled.div`
+  & > * {
+    margin-bottom: 1rem;
+  }
+`;
+
+const FileInput = styled.input`
+  ${visuallyHidden()}
+`;
+
+const UploadSuccess = styled.div`
+  color: ${themeVal('color.success')};
+`;
+
+const UploadWarnings = styled.div`
+  color: ${themeVal('color.info')};
+`;
+
+const UploadError = styled.div`
+  color: ${themeVal('color.danger')};
+`;
+
+interface AoIUploadModalProps {
+  revealed: boolean;
+  onCloseClick: () => void;
+  setFeature: (feature: Feature) => void;
+}
+
+export default function AoIUploadModal({
+  revealed,
+  onCloseClick,
+  setFeature
+}: AoIUploadModalProps) {
+  const {
+    feature,
+    onUploadFile,
+    uploadFileError,
+    uploadFileWarnings,
+    fileInfo,
+    reset
+  } = useCustomAoI();
+  const fileInputRef = useRef<HTMLInputElement>(null);
+
+  const onUploadClick = useCallback(() => {
+    if (fileInputRef.current) fileInputRef.current.click();
+  }, []);
+
+  const onConfirmClick = useCallback(() => {
+    if (!feature) return;
+    setFeature(feature);
+    onCloseClick();
+  }, [feature, setFeature, onCloseClick]);
+
+  useEffect(() => {
+    if (!revealed) reset();
```

Comment:
```suggestion
    if (revealed) reset();
```

I'd recommend resetting on reveal, otherwise any feedback message disappears before you see the modal fade out.